toolchain/supportconfig.py

```diff
@@ -0,0 +1,141 @@
+        List of feature keywords: supportconfig -F
+        Collecting already Rotated Log Files: supportconfig -l
+        """
+        # verify the normal functionality of the supportconfig
+        ret = process.run("supportconfig",
+                          sudo=True,
+                          ignore_status=True)
+        logfile = re.search(r"Log file tar ball: (\S+)\n", ret.stdout).group(1)
+        if not os.path.exists(logfile) or ret.exit_status:
+            self.fail("supportconfig failed to create log file")
+
+        options_to_verify = self.params.get('supportconfig_opt', default="m")
+        feature = self.params.get('feature', default="AUTOFS")
+        for option in options_to_verify:
+            if option == "i":
+                self.log.info("Limiting the Information to a Specific feature")
+                ret = process.run("supportconfig -%s %s" % (option, feature),
+                                  sudo=True,
+                                  ignore_status=True)
+                if not re.search(r"%s...\s*Done" % feature, ret.stdout) \
+                   or ret.exit_status:
+                    self.fail(
+                        "Failed to verify the %s feature using supportconfig" %
+                        feature)
+            elif option == "R":
+                self.log.info(
+                    "Changing default directory path for supportconfig" +
+                    " output files")
+                ret = process.run("supportconfig -%s %s" %
+                                  (option, self.workdir),
+                                  sudo=True,
+                                  ignore_status=True)
+                if self.workdir not in ret.stdout or ret.exit_status:
+                    self.fail(
+                        "Failed to change the supportconfig output" +
+                        " files directory")
+            else:
+                self.log.info("verifying %s option of supportconfig" % option)
+                ret = process.run("supportconfig -%s" % option,
+                                  sudo=True,
+                                  ignore_status=True)
+                if ret.exit_status:
+                    self.fail(
+                        "supportconfig %s option failed with %s exit option" %
+                        (option, ret.exit_status))
+
+    def test_enable_disable_plugins(self):
+        """
+        1.create the /usr/lib/supportconfig/plugins
+        2.copy your script file there
+        3.output will be in the plugin-plugin_name.txt file
+        """
+        plugin_dir = "/usr/lib/supportconfig/plugins"
+        plugin_dir_exists = 1
+        if not os.path.exists(plugin_dir):
+            plugin_dir_exists = 0
+            os.mkdir(plugin_dir)
+        # copy the plugin file
+        plugin_name = '/usr/bin/pstree'
+        process.system("cp %s %s" % (plugin_name, plugin_dir))
+        ret = process.run("supportconfig",
```

Comment:
can't replace process.run() with process.system() , as i am parsing the output from the process.run() to get logile.